test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright 2016-present Datadog, Inc.
+
+package forwarder
+
+import (
+	_ "embed"
+	"fmt"
+	"net"
+	"net/url"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agent"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	awsResources "github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake/fakeintakeparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2vm"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+)
+
+type multiFakeIntakeEnv struct {
+	VM          client.VM
+	Agent       client.Agent
+	Fakeintake1 *client.Fakeintake
+	Fakeintake2 *client.Fakeintake
+}
+
+const (
+	// intakeName should only contains alphanumerical characters to ease pattern matching /etc/hosts
+	intakeName              = "ddintake"
+	connectionResetInterval = 120 // seconds
+	intakeMaxWaitTime       = 5 * time.Minute
+	intakeTick              = 20 * time.Second
+	fakeintake1Name         = "1fakeintake"
+	fakeintake2Name         = "2fakeintake"
+)
+
+//go:embed testfixtures/custom_logs.yaml
+var customLogsConfig string
+
+func multiFakeintakeStackDef(agentOptions ...agentparams.Option) *e2e.StackDefinition[multiFakeIntakeEnv] {
+	return e2e.EnvFactoryStackDef(func(ctx *pulumi.Context) (*multiFakeIntakeEnv, error) {
+		awsEnv, err := awsResources.NewEnvironment(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter1, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake1Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		fiExporter2, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake2Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		agentInstaller, err := agent.NewInstaller(vm, agentOptions...)
+		if err != nil {
+			return nil, err
+		}
+
+		return &multiFakeIntakeEnv{
+			VM:          client.NewPulumiStackVM(vm),
+			Agent:       client.NewPulumiStackAgent(agentInstaller),
+			Fakeintake1: client.NewFakeintake(fiExporter1),
+			Fakeintake2: client.NewFakeintake(fiExporter2),
+		}, nil
+	})
+}
+
+type multiFakeIntakeSuite struct {
+	e2e.Suite[multiFakeIntakeEnv]
+}
+
+func TestMultiFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &multiFakeIntakeSuite{}, multiFakeintakeStackDef())
+}
+
+// SetupSuite waits for both fakeintakes to be ready before running tests.
+func (v *multiFakeIntakeSuite) SetupSuite() {
```

Comment:
Bug: You have to call `e2e.Suite.SetupSuite`.

Reply:
Oh I didn't know that, thanks !